ably/http/http.py

```diff
@@ -158,14 +219,42 @@ def make_request(self, method, url, headers=None, body=None, skip_auth=False, ti
         # log.debug("Prepped: %s" % prepped)
 
         # TODO add timeouts from options here
-        response = self.__session.send(prepped)
+
+        result = self.__session.send(prepped)
+        response = Response(result)
+
+        if response and response.error is not None:
+            if self.handleInvalidToken(response):
+                if self.auth.can_request_token():
+                    self.auth.authorise(force=True)
+                else:
+                    raise AblyException(reason="No way to renew invalid token", status_code=401, code=40140)
+
+            elif self.handleFallback(response):
+                fb = Fallback(Defaults.get_fallback_hosts(self.options))
+                fb_host = ""
+                while fb.should_use_fallback(options, response) and fb_host is not None:
```

Comment:
Should this be a while clause when the call below is recursive?  I may have misunderstood this, but won't this execute until shuld_use_fallback is false and fb_hist is not None, yet it's calling itself so will go through the whole process from the start again?